easybuild/framework/easyconfig/types.py

```diff
@@ -272,22 +275,33 @@ def to_name_version_dict(spec):
         # 2-element list
         if len(spec) == 2:
             res = {'name': spec[0].strip(), 'version': spec[1].strip()}
+        # 3-element list
+        elif len(spec) == 3:
+            res = {'name': spec[0].strip(), 'version': spec[1].strip(), 'hidden': strtobool(spec[2].strip())}
         else:
-            raise EasyBuildError("Can not convert list %s to name and version dict. Expected 2 elements", spec)
+            raise EasyBuildError("Can not convert list %s to toolchain dict. Expected 2 or 3 elements", spec)
 
     elif isinstance(spec, dict):
         # already a dict, check keys
-        if sorted(spec.keys()) == ['name', 'version']:
+        sorted_keys = sorted(spec.keys())
+        if sorted_keys == ['name', 'version'] or sorted_keys == ['hidden', 'name', 'version']:
             res = spec
         else:
-            raise EasyBuildError("Incorrect set of keys in provided dictionary, should be only name/version: %s", spec)
+            raise EasyBuildError("Incorrect set of keys in provided dictionary, should be only name/version/hidden: %s",
+                                 spec)
 
     else:
-        raise EasyBuildError("Conversion of %s (type %s) to name and version dict is not supported", spec, type(spec))
+        raise EasyBuildError("Conversion of %s (type %s) to toolchain dict is not supported", spec, type(spec))
 
     return res
 
 
+def to_name_version_dict(spec):
+    """Deprecated in favor of to_toolchain_dict."""
+    _log.deprecated("to_name_version_dict; use to_toolchain_dict instead.", '2.9')
```

Comment:
@boegel: Are you sure?  The output I get when modifying one of the unit tests is
```
DEPRECATED (since v2.9) functionality used: to_name_version_dict; use to_toolchain_dict instead.; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information
```
This suggests that it is the version since when it is deprecated, no?